File: DALCore/Query/ContactQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;
using DALCore.Models;

namespace DALCore.Query
{
    public class ContactQuery
    {
        public static Expression<Func<CompanyContacts, bool>> GetNameQuery(string name)
        {
            if (string.IsNullOrWhiteSpace(name))
                return null/* TODO Change to default(_) if this is not a reference type */;
            return w => w.ContactNaam.Contains(name) | w.ContactVoornaam.Contains(name);
        }
    }
}

File: DALNEW/Query/ContactQuery.vb
Imports System.Linq.Expressions
Imports BO
Public Class ContactQuery
    Public Shared Function GetNameQuery(name As String) As Expression(Of Func(Of CompanyContacts, Boolean))
        If String.IsNullOrWhiteSpace(name) Then Return Nothing
        Return Function(w) w.ContactNaam.Contains(name) Or w.ContactVoornaam.Contains(name)
    End Function
End Class
